templates/webhook.bin.erb

```diff
@@ -253,14 +253,9 @@ class Server < Sinatra::Base
         # the other one to complete.
         file.flock(File::LOCK_EX)
 
-        if Open3.respond_to?('capture3')
-          stdout, stderr, exit_status = Open3.capture3(command)
-          message = "triggered: #{command}\n#{stdout}\n#{stderr}"
```

Comment:
The problem with the `Open3` calls is that in any case where a user has aggregate webhooks, webhooks that trigger other webhooks, or API calls in an app that trigger a webhook that checks the response code, Gitlab/Github will always return an `HTTP 500` if the command took longer than 10 seconds to execute, which in turn, invalidates the HTTP code the webhook returns.

I've spent a few weeks trying to figure this out and have narrowed it down to `Open3` forcing the API call to hold on to the connection until the command is complete. 

I spent over an hour last night testing logging, etc with the changes and I continued to get logs in the main webhook log. We can make forking a non-default, but as of now, any user that has an r10k environment that takes more than 10 seconds to complete will get an HTTP 500 regardless of what the webhook sends back to Github/Gitlab and potentially Bitbucket (can't find docs on what the timeout is).

Reply:
That is good to hear about the logging.

I believe an immediate failure would still return a 200 with this change. Is there another way to allow those that complete in <10s to return a 500 if the r10k command fails? If not, then an async check around the other fork sections would preserve that.

Reply:
It's possible, though getting more complex. It would look something like this absolutely untested code that *might leave zombies behind!*

```ruby
require 'timeout'

begin  
 Timeout.timeout(10) do
  process = Process.detach(fork { deploy_module(module_name) })
  Process.wait process.pid
  message = "deployed with: #{command}"
 end
rescue Timeout::Error
  message = "forked deploy into background: #{command}"
end
```